src/main/src/main/java/org/geoserver/filters/LoggingFilter.java

```diff
@@ -153,25 +153,25 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
                         // the default encoding for HTTP 1.1
                         encoding = "ISO-8859-1";
                     }
-                    try (InputStream is = hreq.getInputStream()) {
-
-                        Charset charset = Charset.defaultCharset();
-                        try {
-                            charset = Charset.forName(encoding);
-                        } catch (IllegalCharsetNameException icn) {
-                            logger.info(
-                                    "Request character set not recognized, defaulting to ISO-8859-1");
-                        }
-                        float maxBytesPerCharacter = charset.newEncoder().maxBytesPerChar();
-                        int byteSize = (int) (requestLogBufferSize * (double) maxBytesPerCharacter);
-                        byte[] reqCharacters = new byte[byteSize];
-                        BufferedInputStream bufferedStream = new BufferedInputStream(is);
-                        bufferedStream.mark(byteSize);
-                        bufferedStream.read(reqCharacters, 0, byteSize);
-                        body = new String(reqCharacters, encoding).trim();
-                        bufferedStream.reset();
-                        req = new BufferedRequestWrapper(hreq, encoding, bufferedStream);
+                    InputStream is = hreq.getInputStream();
```

Comment:
Surprised PMD is not complaining (well the build is undergoing, it still might).
In any case, I'd add a comment explaining why this is not using a try-with-resource (not to me, here, I know why, but in the code, as a comment, for the future developer that might read and wonder).

Reply:
okay something like:
```
// Stream opened for BufferedRequestWrapper use - do not close
```

Reply:
I'd go for 

```
// The HTTPServletResponse stream should not be closed (Tomcat would complain), so avoid usage or try-with-resources
```